src/ert/scheduler/job.py

```diff
@@ -67,6 +73,7 @@ async def __call__(
     ) -> None:
         await start.wait()
         await sem.acquire()
+        cancel_task: Optional[asyncio.Task[None]] = None
```

Comment:
Could we use `asyncio.wait_for` instead of creating our own cancel task?

Reply:
It looks like `wait_for` is non-trivial because of the ugly:
```
            while not self.returncode.done():
                await asyncio.sleep(0.01)
            returncode = await self.returncode
```
which we then would have to wrap in yet another function (?)